src/components/oculus-touch-controls.js

```diff
@@ -139,26 +174,38 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
     this.el.setAttribute('obj-model', {obj: objUrl, mtl: mtlUrl});
   },
 
-  injectTrackedControls: function () {
+  injectTrackedControls: function (gamepad) {
     var el = this.el;
     var data = this.data;
     var isRightHand = data.hand === 'right';
-
-    // since each hand is named differently, avoid enumeration
+    // Inject with specific gamepad id, if provided. This works around a temporary issue
+    // where Chromium uses `Oculus Touch (Right)` but Firefox uses `Oculus Touch (right)`.
     el.setAttribute('tracked-controls', {
```

Comment:
might want to check out these Gamepad librarys I wrote; maybe there are some parts you can reuse.

* https://github.com/MozVR/oculus/blob/5aa680f/assets/js/gamepad.js
* https://github.com/MozVR/oculus/blob/e73ea2c/assets/js/common.js#L107
* https://github.com/MozVR/gamepad-plus
* https://github.com/MozVR/gamepad-plus/blob/b3cc035/demo.js